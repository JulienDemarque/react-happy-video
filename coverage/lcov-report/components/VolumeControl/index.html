<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/VolumeControl</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/VolumeControl
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="83.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 83%;"></div><div class="cover-empty" style="width:17%;"></div></div></td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 05 2018 20:47:31 GMT+0200 (IST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
