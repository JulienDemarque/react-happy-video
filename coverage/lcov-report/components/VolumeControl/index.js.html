<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/VolumeControl/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/VolumeControl</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">81.82% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Container, VolumeBtn, VolumeBar } from './styles';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
	faVolumeDown,
	faVolumeMute,
	faVolumeUp
} from '@fortawesome/free-solid-svg-icons';
&nbsp;
class VolumeControl extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			viewVolumeBar: false
		};
		this.toggleVolumeBar = this.toggleVolumeBar.bind(this);
	}
&nbsp;
	toggleVolumeBar() {
		this.setState(state =&gt; ({ viewVolumeBar: !state.viewVolumeBar }));
	}
&nbsp;
	render() {
		const { volume } = this.props;
		let Icon;
		<span class="missing-if-branch" title="if path not taken" >I</span>if (volume === '0') {
<span class="cstat-no" title="statement not covered" >			Icon = &lt;FontAwesomeIcon icon={faVolumeMute} /&gt;;</span>
		} else <span class="missing-if-branch" title="if path not taken" >I</span>if (volume &gt; 50) {
<span class="cstat-no" title="statement not covered" >			Icon = &lt;FontAwesomeIcon icon={faVolumeUp} /&gt;;</span>
		} else {
			Icon = &lt;FontAwesomeIcon icon={faVolumeDown} /&gt;;
		}
&nbsp;
		return (
			&lt;Container&gt;
				{this.state.viewVolumeBar ? (
					&lt;VolumeBar
						value={this.props.volume}
						onChange={this.props.handleVolumeChange}
					/&gt;
				) : (
					''
				)}
				&lt;VolumeBtn onClick={this.toggleVolumeBar}&gt;{Icon}&lt;/VolumeBtn&gt;
			&lt;/Container&gt;
		);
	}
}
&nbsp;
export default VolumeControl;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 05 2018 20:47:31 GMT+0200 (IST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
