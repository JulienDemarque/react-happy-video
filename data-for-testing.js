export const ProgressBarProps = {
	colorgiven: '#3b3346',
	progress: 0.2
};

export const SpeedControlProps = {
	speed: 1
};

export const TimeDisplayProps = {
	time: 20,
	duration: 300
};

export const VideoPlayerProps = {
	width: '100%',
	source: 'www.mymockurl.com/mymockvideo.mp4',
	color: '#3b3346'
};

export const VolumeControlProps = {
	volume: 0.5
};
