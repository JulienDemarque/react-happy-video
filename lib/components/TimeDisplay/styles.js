import styled from 'styled-components';

export const SubContainer = styled.div`
	display: flex;
	align-items: center;
	padding: 0 20px;
`;
