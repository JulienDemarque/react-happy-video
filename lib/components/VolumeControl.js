import React from 'react';
import {
	Container,
	VolumeBtn,
	VolumeBar
} from '../styles/components/volume-control';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
	faVolumeDown,
	faVolumeMute,
	faVolumeUp
} from '@fortawesome/free-solid-svg-icons';

class VolumeControl extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			viewVolumeBar: false
		};
		this.toggleVolumeBar = this.toggleVolumeBar.bind(this);
	}

	toggleVolumeBar() {
		this.setState(state => ({ viewVolumeBar: !state.viewVolumeBar }));
	}

	render() {
		let Icon;
		if (this.props.volume === '0') {
			Icon = <FontAwesomeIcon icon={faVolumeMute} />;
		} else if (this.props.volume > 50) {
			Icon = <FontAwesomeIcon icon={faVolumeUp} />;
		} else {
			Icon = <FontAwesomeIcon icon={faVolumeDown} />;
		}

		return (
			<Container>
				{this.state.viewVolumeBar ? (
					<VolumeBar
						value={this.props.volume}
						onChange={this.props.handleVolumeChange}
					/>
				) : (
					''
				)}
				<VolumeBtn onClick={this.toggleVolumeBar}>{Icon}</VolumeBtn>
			</Container>
		);
	}
}

export default VolumeControl;
