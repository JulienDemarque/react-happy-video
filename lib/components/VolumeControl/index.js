import React from 'react';
import { Container, VolumeBtn, VolumeBar } from './styles';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
	faVolumeDown,
	faVolumeMute,
	faVolumeUp
} from '@fortawesome/free-solid-svg-icons';

class VolumeControl extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			viewVolumeBar: false
		};
		this.toggleVolumeBar = this.toggleVolumeBar.bind(this);
	}

	toggleVolumeBar() {
		this.setState(state => ({ viewVolumeBar: !state.viewVolumeBar }));
	}

	render() {
		const { volume } = this.props;
		let Icon;
		if (volume === '0') {
			Icon = <FontAwesomeIcon icon={faVolumeMute} />;
		} else if (volume > 50) {
			Icon = <FontAwesomeIcon icon={faVolumeUp} />;
		} else {
			Icon = <FontAwesomeIcon icon={faVolumeDown} />;
		}

		return (
			<Container>
				{this.state.viewVolumeBar ? (
					<VolumeBar
						value={this.props.volume}
						onChange={this.props.handleVolumeChange}
					/>
				) : (
					''
				)}
				<VolumeBtn onClick={this.toggleVolumeBar}>{Icon}</VolumeBtn>
			</Container>
		);
	}
}

export default VolumeControl;
