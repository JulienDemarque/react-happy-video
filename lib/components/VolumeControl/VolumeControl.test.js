import React from 'react';
import { shallow } from 'enzyme';
import VolumeControl from './index';
import { VolumeBar, VolumeBtn } from './styles';
import { VolumeControlProps } from '../../../data-for-testing';

const { volume } = VolumeControlProps;

describe('VolumeControl Component', () => {
	let wrapper;
	const mockHandleVolumeChange = jest.fn();

	beforeEach(() => {
		wrapper = shallow(
			<VolumeControl
				volume={volume}
				handleVolumeChange={mockHandleVolumeChange}
			/>
		);
	});

	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('VolumeBar calls handleVolumeChange on change', () => {
		wrapper.setState({ viewVolumeBar: true });
		wrapper.find(VolumeBar).simulate('change', { target: { value: 0.7 } });
		expect(mockHandleVolumeChange.mock.calls.length).toBe(1);
	});

	it('VolumeBtn updates state on change', () => {
		wrapper.setState({ viewVolumeBar: false });
		wrapper.find(VolumeBtn).simulate('click');
		expect(wrapper.state().viewVolumeBar).toBe(true);
	});
});
