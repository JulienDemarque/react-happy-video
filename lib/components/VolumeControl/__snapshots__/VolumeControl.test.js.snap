// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeControl Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VolumeControl
    handleVolumeChange={[MockFunction]}
    volume={0.5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "",
        <ForwardRef
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f027",
                  "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
                ],
                "iconName": "volume-down",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f027",
                  "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
                ],
                "iconName": "volume-down",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "",
            "fixedWidth": false,
            "flip": null,
            "icon": Object {
              "icon": Array [
                384,
                512,
                Array [],
                "f027",
                "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
              ],
              "iconName": "volume-down",
              "prefix": "fas",
            },
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
	background-color: transparent;
	border: none;
	font-size: 22px;
	padding: 0;
	outline: none;
	&:hover {
		cursor: pointer;
	}
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
	display: flex;
	position: relative;
	width: 30px;
	padding-left: 20px;
	padding-right: 10px;
	text-align: left;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "",
          <ForwardRef
            onClick={[Function]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f027",
                    "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
                  ],
                  "iconName": "volume-down",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    384,
                    512,
                    Array [],
                    "f027",
                    "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
                  ],
                  "iconName": "volume-down",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": Object {
                "icon": Array [
                  384,
                  512,
                  Array [],
                  "f027",
                  "M215.03 72.04L126.06 161H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V89.02c0-21.47-25.96-31.98-40.97-16.98zm123.2 108.08c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 229.28 336 242.62 336 257c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.87z",
                ],
                "iconName": "volume-down",
                "prefix": "fas",
              },
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
	background-color: transparent;
	border: none;
	font-size: 22px;
	padding: 0;
	outline: none;
	&:hover {
		cursor: pointer;
	}
",
              ],
            },
            "displayName": "styled.button",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "button",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
	display: flex;
	position: relative;
	width: 30px;
	padding-left: 20px;
	padding-right: 10px;
	text-align: left;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
